craft_parts/data/schema/parts.json

```diff
@@ -0,0 +1,423 @@
+                    "type": "array",
+                    "description": "deb types to enable.  Defaults to [deb, deb-src].",
+                    "minItems": 1,
+                    "uniqueItems": true,
+                    "items": {
+                        "type": "string",
+                        "enum": [
+                            "deb",
+                            "deb-src"
+                        ]
+                    }
+                },
+                "components": {
+                    "type": "array",
+                    "minItems": 0,
+                    "uniqueItems": true,
+                    "items": {
+                        "type": "string",
+                        "description": "Deb repository components to enable, e.g. 'main, multiverse, unstable'"
+                    }
+                },
+                "key-id": {
+                    "type": "string",
+                    "description": "GPG key identifier / fingerprint. May be used to identify key file in <project>/snap/keys/<key-id>.asc",
+                    "pattern": "^[a-zA-Z0-9-_]*$"
+                },
+                "key-server": {
+                    "type": "string",
+                    "description": "GPG keyserver to use to fetch GPG <key-id>, e.g. 'keyserver.ubuntu.com'. Defaults to keyserver.ubuntu.com if key is not found in project."
+                },
+                "suites": {
+                    "type": "array",
+                    "minItems": 1,
+                    "uniqueItems": true,
+                    "items": {
+                        "type": "string",
+                        "description": "Deb repository suites to enable, e.g. 'xenial-updates, xenial-security')."
+                    }
+                },
+                "url": {
+                    "type": "string",
+                    "description": "Deb repository URL, e.g. 'http://archive.canonical.com/ubuntu'."
+                }
+            },
+            "required": [
+                "type",
+                "components",
+                "key-id",
+                "suites",
+                "url"
+            ],
+            "validation-failure": "{!r} is not properly configured deb repository"
+        },
+        "apt-ppa": {
+            "type": "object",
+            "description": "PPA repository",
+            "additionalProperties": false,
+            "properties": {
+                "type": {
+                    "type": "string",
+                    "enum": [
+                        "apt"
+                    ]
+                },
+                "ppa": {
+                    "type": "string",
+                    "description": "ppa path: e.g. 'canonical-kernel-team/unstable'"
+                }
+            },
+            "required": [
+                "type",
+                "ppa"
+            ],
+            "validation-failure": "{!r} is not properly configured PPA repository"
+        },
+        "system-username-scope": {
```

Comment:
this only applies to snaps

Reply:
There are other definitions that are not used, will remove them all.